templates/common/infra/bicep/core/database/postgresql/flexibleserver.bicep

```diff
@@ -29,6 +30,20 @@ resource postgresServer 'Microsoft.DBforPostgreSQL/flexibleServers@2022-12-01' =
     highAvailability: {
       mode: 'Disabled'
     }
+    authConfig: {
+      activeDirectoryAuth: activeDirectoryAuth
+      passwordAuth: (administratorLoginPassword == null) ? 'Disabled' : 'Enabled'
+    }
+    backup: {
+      backupRetentionDays: 7
+      geoRedundantBackup: 'Disabled'
+    }
+    maintenanceWindow: {
+      customWindow: 'Disabled'
+      dayOfWeek: 0
+      startHour: 0
+      startMinute: 0
+    }
```

Comment:
These should be params with defaults so user can mod if they want

Reply:
I've updated the PR with params.

Where the setting name wasn't as clear like `customWindow` I made the parameter `maintenanceWindowCustomWindow`, but I'm wondering about doing something more like 'maintenanceWindow_customWindow` to indicate the relationship more clearly.

Thoughts?

Reply:
Could you look into: https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/parameters#objects-as-parameters